### PR TITLE
mv `on_schema_change` tests -> "adapter zone"

### PR DESCRIPTION
**Pull request by jtcohen6**
follow-up to https://github.com/dbt-labs/dbt-core/pull/6330

### Description

In the previous conversion PR, @VersusFacit made a reasonable point:
> Oh, do we want these imported into any repos? Incremental tests have so many caveats, I'm unsure a direct import is valuable in this case.
> 
> All I did was convert the postgres__ tests to ordinary tests to live in `tests/functional`

As it stands, that these tests have _already_ been living in those other repos, and they were originally written (in https://github.com/dbt-labs/dbt-core/pull/3387) to be cross-adapter compatible. Then, they were copy-pasted everywhere, as part of the Great Adapter Split, in fall 2021 / the lead-up to v1.0.

By moving this test into the "adapter zone" (a.k.a. `dbt-tests-adapter`), we can inherit it for use in all our adapter repos, and delete a LOT of copy-pasted code.

While here:
- Replace the hardcoded `string_type` with `dbt.type_string()` (added in v1.2!)
- Refactor the test setup _very slightly_ so it's easier to extend/repurpose (relevant to `dbt-bigquery`), without having to run existing test functions multiple times
- Remove the test fixtures, since they were actually unused in the test
- Do a little bit of file renaming / reorg within `dbt-tests-adapter` (which is v1 distributed software!)

### Adapter PRs
- https://github.com/dbt-labs/dbt-redshift/pull/269
- https://github.com/dbt-labs/dbt-snowflake/pull/398
- https://github.com/dbt-labs/dbt-bigquery/pull/469
- https://github.com/dbt-labs/dbt-spark/pull/593

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- ~I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)~

### CODE REVIEW

**Review by ChenyuLInx — APPROVED** [review]
LGTM!

File: tests/adapter/dbt/tests/adapter/incremental/fixtures.py [hunk 1]
```diff
@@ -361,35 +303,3 @@
 
 from source_data
 """
-
-#
-# Tests
-#
-
-_TESTS__SELECT_FROM_INCREMENTAL_IGNORE = """
```
Comment (VersusFacit) [inline tests/adapter/dbt/tests/adapter/incremental/fixtures.py:369]: Clearly from my more naive days of test conversions :p 

That said, I do recall thinking "should we keep these?" I know they live in the code history and don't necessary need to exist in current code. More a "principles" question: if there are seemingly extraneously models in tests, do we nix them with cautious abandon or seek to add tests to include them?

**Review by VersusFacit — APPROVED** [review]
Barring the merge conflicts, and my lil' comment, looks good.

Thanks for introducing me to that neat `dbt.type_string()`.

File: tests/adapter/dbt/tests/adapter/incremental/fixtures.py (hunk 1 above)
Comment (jtcohen6) [reply tests/adapter/dbt/tests/adapter/incremental/fixtures.py:369]: As a general rule: If it's not being actively used in one of our existing tests, we probably don't need it. (There are rare exceptions, where we've added a new fixture/combination with the _intent_ of testing, but completely forgot! I just found an instance where I'd made this mistake a ~while ago: https://github.com/dbt-labs/dbt-bigquery/pull/41)

Longer answer: Intent is a useful input, to the extent we can infer it. In this case, I remember the history as being: our colleague Matt W added this feature (`on_schema_change`) back in 2021, and he added integration tests for it by copying + adapting some other recent integration tests (for `--store-failures`). The "tests" component was important for the original tests, but not necessary for these adapted ones.

Comment (VersusFacit) [reply tests/adapter/dbt/tests/adapter/incremental/fixtures.py:368]: Appreciate the history lesson! That makes much more sense and I'll run with your heuristic in the future.

**Pull request merged**